packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -274,10 +276,22 @@ export class RunningSummarizer implements IDisposable {
             case MessageType.SummaryNack:
                 return false;
             default:
-                return true;
+                return isRuntimeMessage(op)
+                    || RunningSummarizer.nonRuntimeOpCanTriggerSummary(this.configuration, this.heuristicData);
```

Comment:
I think this method's contract refers to the op itself (specifically the type) and not to the surrounding context. So, it answers the question 'Can this type of op trigger a summary?' not 'Should this op trigger a summary right now?'. This change makes it answer the latter. I don't have any strong feelings about it, but maybe the comment needs to be updated.